### DISCUSSION TITLE
R - Menghitung jumlah dalam rentang berdasarkan kondisi.

### DISCUSSION THREAD

**Discussion by Hertzspr** [teamalgoritma/community#62] [Q&A] [answered]
Selamat siang, boleh minta bantuannya kira-kira apakah di R bisa membuat agregasi data yang dapat menghitung jumlah dalam rentang berdasarkan kondisi? Apakah ada fungsi yang bisa digunakan dengan praktis atau perlu membuat fungsi sendiri?

Spesifiknya, bagaimana menghitung total berapa hari/ berapa streams: 
1. Berdasarkan muncul penambahan nilai stream pertama sejak ada kolom song yang tidak NA, 
2. Hingga kolom streams bernilai 0 berturut-turut selama 7 hari,

csv terlampir
[syh_timelines_snapshot.csv](https://github.com/teamalgoritma/community/files/8537599/syh_timelines_snapshot.csv)

**Comment by inytss** [comment, marked as answer]
Dear @Hertzspr , 
Berdasarkan data yang Bapak miliki, saya coba asumsikan seperti ini:

- tabel tersebut mengasumsikan 1 jenis lagu saja yang dilakukan aktifitas streaming selama tanggal 9 Jan 2020 - 8 Mar 2020. 

Maka ketika kita ingin mencari tahu jumlah stream untuk masing-masing tanggal yang ada dapat melakukan group_by berdasarkan tanggal dan menjumlahkan nilai pada kolom `streams`. Namun jika spesifik ingin mengetahui tanggal berapa saja yang streamnya = 0, maka bisa ditambahkan dengan `filter()` jumlah stream yang = 0  Berikut contoh codenya
```
df %>% 
  mutate(date = ymd(date)) %>% 
  group_by(date) %>% 
  summarise(sum_stream = sum(streams)) %>% 
  filter(sum_stream == 0)
```

Saran, jika ingin lebih bisa melihat pergerakan stream masing-masing tanggal menggunakan line chart supaya terlihat di tanggal berapa saja lagu tidak ada yang melakukan streaming.

Demikian jawaban dari saya, semoga membantu :)
